3d/navmesh/navmesh.gd

```
@@ -3,85 +3,83 @@ extends Navigation
 const SPEED = 10.0
 
 var camrot = 0.0
-
-var begin = Vector3()
-var end = Vector3()
 var m = SpatialMaterial.new()
 
 var path = []
 var draw_path = true
 
-func _ready():
-	set_process_input(true)
-	m.flags_unshaded = true
-	m.flags_use_point_size = true
-	m.albedo_color = Color.white
-
-
-func _process(delta):
-	if path.size() > 1:
-		var to_walk = delta * SPEED
-		var to_watch = Vector3.UP
-		while to_walk > 0 and path.size() >= 2:
-			var pfrom = path[path.size() - 1]
-			var pto = path[path.size() - 2]
-			to_watch = (pto - pfrom).normalized()
-			var d = pfrom.distance_to(pto)
-			if d <= to_walk:
-				path.remove(path.size() - 1)
-				to_walk -= d
-			else:
-				path[path.size() - 1] = pfrom.linear_interpolate(pto, to_walk / d)
-				to_walk = 0
-
-		var atpos = path[path.size() - 1]
-		var atdir = to_watch
-		atdir.y = 0
-
-		var t = Transform()
-		t.origin = atpos
-		t = t.looking_at(atpos + atdir, Vector3.UP)
-		get_node("RobotBase").set_transform(t)
-
-		if path.size() < 2:
-			path = []
-			set_process(false)
-	else:
-		set_process(false)
+onready var robot  = get_node("RobotBase")
+onready var camera = get_node("CameraBase/Camera")
 
+func _ready():
+    set_process_input(true)
+    m.flags_unshaded = true
+    m.flags_use_point_size = true
+    m.albedo_color = Color.white
```

Tabs are used for indentation in GDScript.

---

Sorry, I hadn't noticed my script editor had silently converted the whole thing to spaces..